python-checks/src/main/java/org/sonar/python/checks/NewSymbolsAnalyzer.java

```diff
@@ -36,7 +37,10 @@
     List<AstNode> methods = classDef.getFirstChild(PythonGrammar.SUITE).getDescendants(PythonGrammar.FUNCDEF);
 
     for (AstNode method : methods) {
-      addFieldsInMethod(method);
+      // ignore any nested class
+      if (classDef == method.getFirstAncestor(PythonGrammar.CLASSDEF)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)